server/channels/store/sqlstore/thread_store.go

```diff
@@ -134,11 +134,16 @@ func (s *SqlThreadStore) getTotalThreadsQuery(userId, teamId string, opts model.
 		})
 
 	if teamId != "" {
-		query = query.
-			Where(sq.Or{
-				sq.Eq{"Threads.ThreadTeamId": teamId},
-				sq.Eq{"Threads.ThreadTeamId": ""},
-			})
+		if !opts.ExcludeDirect {
+			query = query.Where(
+				sq.Or{
+					sq.Eq{"Threads.ThreadTeamId": teamId},
+					sq.Eq{"Threads.ThreadTeamId": ""},
+				},
+			)
+		} else {
+			query = query.Where(sq.Eq{"Threads.ThreadTeamId": teamId})
+		}
```

Comment:
True although most cases and default behavior is that the prop is set to false. But I can change it if needed or is wanted

Reply:
nit: I see what @Willyfrog -- ExcludeDirect is a negative variable name, meaning not to include direct message channels.  Adding the (!) means to NOT not include direct message channels, or to include direct messages.  I understand what it's doing, but it does get confusing.

Also, the title is to "optionally **exclude threads** in direct messages", `ExcludeDirect` got me to think that we're excluding direct messages more than excluding threads.

Alternatively, do consider `includeThreads` and default to `true` but that could be confusing with threadsOnly?  (0/5)

Same goes with the rest of the if condition.

Just my $0.02.

Reply:
Really? What about setting the variable name to `shouldExcludeThreadsInDirectAndGroupMessages` I think that works absolutely fantastic here, but hey, then it does not match the query string, should we also update the query string?

both of you did say is non-blocking but you guys clearly have an opinion on this, I think is best if we agree on something before merging.

Btw @rahimrahman in my opinion `includeThreads` alone would be extremely confusing as this whole endpoint is to fetch threads 🤣🤣

Reply:
Or we can also call it `shouldIncludeThreadsInDirectAndGroupMessages` and default it to true, but then the query string is also a bit too long

Reply:
I see what you're saying.  And it is my oversight that I completely missed that excludeDirect is a query string ONLY in thread API.  Clearly, I shouldn't be making opinion on variable naming until I fully grasp the affect of the changes you're making. 😆 

Regardless, I did have an opinion about double negative use in variable name  + if condition.  if !opts.excludeDirect really means if not not include direct message or if not exclude direct message.

I guess knowing what I know now, excludeDirect as a query string is fine, but !opts.excludeDirect means "if not excluding direct messages".  And the way @Willyfrog is acceptable where we reverse it.

Thus non-blocking, as you can see, my opinion was somewhat skewed to the wrong understanding.

Reply:
changes were made

Reply:
> both of you did say is non-blocking but you guys clearly have an opinion on this, I think is best if we agree on something before merging.

Sorry if that confused you. I think it is better, but it might be a matter of opinion and I rather not enforce my opinions on readability on others.